stream-chat-android-ui-components/src/main/kotlin/io/getstream/chat/android/ui/viewmodel/messages/MessageListViewModel.kt

```diff
@@ -168,16 +172,30 @@ public class MessageListViewModel(
      * Initializes the full message list state conversion and collection.
      */
     init {
-        val listState = messageListController.listState
-            .onSubscription { State.Loading }
-            .map {
-                if (it.isLoading) {
-                    State.Loading
-                } else {
-                    State.Result(it.toMessageListItemWrapper())
-                }
-            }.asLiveData()
-        stateMerger.addSource(listState) { stateMerger.value = it }
+        viewModelScope.launch {
+            val listState = messageListController.listState.combine(messageListController.mode) { listState, mode ->
+                Pair(listState, mode)
+            }
+                // TODO - Think of a better solution once we have more capacity to do larger refactors
+                // Due to the way we combine upstream flows in MessageListController, we get unnecessary multiple
+                // emissions at the start of a collecting a new state. These multiple emissions happen within a
+                // millisecond of one another and can force the RecyclerView adapter to skip actions when loading a
+                // thread.
+                .debounce(5)
```

Comment:
I'm aware debouncing something like this is not the best way to go forwards, but we cannot expend much more time on this task currently and in all honesty the best course of action might be a larger refactor of our state handling and the `MessageListScrollHelper`.

If everything done in this PR turns out to address the issues in a more reliable fashion, we should probably spin up a task for a larger refactor and add it to the backlog.